tests/test_minimal/test_metadata_schema.py

```diff
@@ -0,0 +1,254 @@
+                "name": "Fluorescence",
+                "fluorescence_chan1_plane0": dict(),  # Value Unchecked
+                "FluorescenceChan1Plane0": dict(),
+                "FluorescenceChan1Plane1": dict(
+                    raw=dict(),
+                ),
+            },
+            DFOverF={
+                "name": "DfOverF",
+                "df_chan1_plane0": dict(),  # Value Unchecked
+                "DFChan1Plane0": dict(),
+                "DFChan1Plane1": dict(
+                    raw=dict(),
+                ),
+            },
+            SegmentationImages={
+                "name": "SegmentationImages",
+                "segmentation_chan1_plane0": dict(),  # Value Unchecked
+                "SegmentationChan1Plane0": dict(),
+                "SegmentationChan1Plane1": dict(
+                    raw=dict(),
+                ),
+            },
+        ),
+    )
+
+    expected_errors = [
+        dict(
+            message="'fluorescence_chan1_plane0' does not match any of the regexes: '^(?!name$)[a-zA-Z0-9]+$'",
+            path="$.Ophys.Fluorescence",
+        ),
+        dict(
+            message="'df_chan1_plane0' does not match any of the regexes: '^(?!name$)[a-zA-Z0-9]+$'",
+            path="$.Ophys.DFOverF",
+        ),
+        dict(
+            message="'segmentation_chan1_plane0' does not match any of the regexes: '^(?!(name|description)$)[a-zA-Z0-9]+$'",
+            path="$.Ophys.SegmentationImages",
+        ),
+        dict(
+            message="{} does not have enough properties",
+            path="$.Ophys.Fluorescence.FluorescenceChan1Plane0",
+        ),
+        dict(
+            message="{} does not have enough properties",
+            path="$.Ophys.DFOverF.DFChan1Plane0",
+        ),
+        dict(
+            message="{} does not have enough properties",
+            path="$.Ophys.SegmentationImages.SegmentationChan1Plane0",
+        ),
+        dict(
+            message="'name' is a required property",
+            path="$.Ophys.Fluorescence.FluorescenceChan1Plane1.raw",
+        ),
+        dict(
+            message="'name' is a required property",
+            path=f"$.Ophys.DFOverF.DFChan1Plane1.raw",
+        ),
+        dict(
+            message="'name' is a required property",
+            path="$.Ophys.SegmentationImages.SegmentationChan1Plane1.raw",
+        ),
+        dict(
+            message="'description' is a required property",
+            path="$.Ophys.Fluorescence.FluorescenceChan1Plane1.raw",
+        ),
+        dict(
+            message="'description' is a required property",
+            path="$.Ophys.DFOverF.DFChan1Plane1.raw",
+        ),
+    ]
+
+    validator = jsonschema.Draft7Validator(metadata_schema)
+
+    errors = [{"message": error.message, "path": error.json_path} for error in validator.iter_errors(metadata)]
+
+    def sorting_fn(o):
+        return o["message"]
+
+    errors.sort(key=sorting_fn)
+    expected_errors.sort(key=sorting_fn)
```

Comment:
These don't need to be sorted if we're using the strategy of iterating over elements asking if they are `in` the list, do we?

```suggestion
```

Reply:
Ah duh. Thanks!